src/__tests__/hybrid-assignment-cache.test.tsx

```diff
@@ -0,0 +1,63 @@
+import { AsyncStorageAssignmentCache } from '../cache/async-storage-assignment-cache';
+import HybridAssignmentCache from '../cache/hybrid-assignment-cache';
+import SimpleAssignmentCache from '../cache/simple-assignment-cache';
+
+describe('HybridStorageAssignmentCache', () => {
+  let asyncStorageCache: AsyncStorageAssignmentCache;
+  let simpleCache: SimpleAssignmentCache;
+  let hybridCache: HybridAssignmentCache;
+
+  beforeEach(() => {
+    asyncStorageCache = new AsyncStorageAssignmentCache('test');
+    simpleCache = new SimpleAssignmentCache();
+    hybridCache = new HybridAssignmentCache(simpleCache, asyncStorageCache);
+  });
+
+  it('has should return false if cache is empty', async () => {
+    const cacheKey = {
+      subjectKey: 'subject-1',
+      flagKey: 'flag-1',
+      allocationKey: 'allocation-1',
+      variationKey: 'control',
+    };
+    await hybridCache.init();
+    expect(hybridCache.has(cacheKey)).toBeFalsy();
+  });
+
+  it('has should return true if cache key is present', async () => {
+    const cacheKey = {
+      subjectKey: 'subject-1',
+      flagKey: 'flag-1',
+      allocationKey: 'allocation-1',
+      variationKey: 'control',
+    };
+    await hybridCache.init();
+    expect(hybridCache.has(cacheKey)).toBeFalsy();
+    expect(simpleCache.has(cacheKey)).toBeFalsy();
+    hybridCache.set(cacheKey);
+    expect(hybridCache.has(cacheKey)).toBeTruthy();
+    expect(simpleCache.has(cacheKey)).toBeTruthy();
+  });
+
+  it('should populate asyncStorageCache from simpleCache', async () => {
+    const key1 = {
+      subjectKey: 'subject-1',
+      flagKey: 'flag-1',
+      allocationKey: 'allocation-1',
+      variationKey: 'control',
+    };
+    const key2 = {
+      subjectKey: 'subject-2',
+      flagKey: 'flag-2',
+      allocationKey: 'allocation-2',
+      variationKey: 'control',
+    };
+    expect(simpleCache.has(key1)).toBeFalsy();
+    asyncStorageCache.set(key1);
+    asyncStorageCache.set(key2);
+    await hybridCache.init();
+    expect(simpleCache.has(key1)).toBeTruthy();
+    expect(simpleCache.has(key2)).toBeTruthy();
+    expect(simpleCache.has({ ...key1, allocationKey: 'foo' })).toBeFalsy();
+  });
```

Comment:
I think we should also test what happens if the subject key and flag key and allocation key stay the same but variation changes. In that situation, we'd expect `has()` to return false, and then after setting the new one expect `has()` for the old variation to be false. This is a pattern we do elsewhere so we log if a user is assigned multiple variations

Reply:
Added!